source/base/polynomials_vector_anisotropic.cc

```diff
@@ -0,0 +1,339 @@
+                                                    [d2][d3];
+
+      for (unsigned int i = 0; i < p_fourth_derivatives.size(); ++i)
+        for (unsigned int d1 = 0; d1 < dim; ++d1)
+          for (unsigned int d2 = 0; d2 < dim; ++d2)
+            for (unsigned int d3 = 0; d3 < dim; ++d3)
+              for (unsigned int d4 = 0; d4 < dim; ++d4)
+                fourth_derivatives[lexicographic_to_hierarchic[i + d * n_sub]]
+                                  [d][(d1 + d) % dim][(d2 + d) % dim]
+                                  [(d3 + d) % dim][(d4 + d) % dim] =
+                                    p_fourth_derivatives[renumber_aniso[d][i]]
+                                                        [d1][d2][d3][d4];
+    }
+}
+
+
+
+template <int dim>
+std::string
+PolynomialsVectorAnisotropic<dim>::name() const
+{
+  return "VectorAnisotropic";
+}
+
+
+
+template <int dim>
+unsigned int
+PolynomialsVectorAnisotropic<dim>::n_polynomials(
+  const unsigned int normal_degree,
+  const unsigned int tangential_degree)
+{
+  return dim * (normal_degree + 1) *
+         Utilities::pow(tangential_degree + 1, dim - 1);
+}
+
+
+
+template <int dim>
+unsigned int
+PolynomialsVectorAnisotropic<dim>::get_tangential_degree() const
+{
+  return tangential_degree;
+}
+
+
+
+template <int dim>
+unsigned int
+PolynomialsVectorAnisotropic<dim>::get_normal_degree() const
+{
+  return normal_degree;
+}
+
+
+
+template <int dim>
+std::unique_ptr<TensorPolynomialsBase<dim>>
+PolynomialsVectorAnisotropic<dim>::clone() const
+{
+  return std::make_unique<PolynomialsVectorAnisotropic<dim>>(*this);
+}
+
+
+
+template <int dim>
+std::vector<Point<dim>>
+PolynomialsVectorAnisotropic<dim>::get_polynomial_support_points() const
+{
+  Assert(dim > 0 && dim <= 3, ExcImpossibleInDim(dim));
+  const Quadrature<1> tangential(
+    tangential_degree == 0 ?
+      static_cast<Quadrature<1>>(QMidpoint<1>()) :
+      static_cast<Quadrature<1>>(QGaussLobatto<1>(tangential_degree + 1)));
+  const Quadrature<1> normal(
+    normal_degree == 0 ?
+      static_cast<Quadrature<1>>(QMidpoint<1>()) :
+      static_cast<Quadrature<1>>(QGaussLobatto<1>(normal_degree + 1)));
+  const QAnisotropic<dim> quad =
+    (dim == 1 ? QAnisotropic<dim>(normal) :
+                (dim == 2 ? QAnisotropic<dim>(normal, tangential) :
+                            QAnisotropic<dim>(normal, tangential, tangential)));
+  std::vector<std::vector<Point<1>>> points_aniso;
+  points_aniso =
+    create_anisotropic_support_points(dim, normal_degree, tangential_degree);
+  const unsigned int      n_sub = polynomial_space.n();
+  std::vector<Point<dim>> points(dim * n_sub);
+  points.resize(n_polynomials(normal_degree, tangential_degree));
+  std::vector<unsigned int> indices_points_anisotropic(dim);
+  unsigned int              renumber_aniso_ind;
+  for (unsigned int d = 0; d < dim; ++d)
+    for (unsigned int i = 0; i < n_sub; ++i)
+      {
+        renumber_aniso_ind = renumber_aniso[d][i];
```

Comment:
This is what I wrote above: I would create the variable here, and possibly use a more descriptive name
```suggestion
        unsigned int renumbered_index = renumber_aniso[d][i];
```